File: JavaScript-Reverse/Crypto/PBKDF2/pbkdf2_demo.py
# -*- coding:utf-8 -*-
# @Author: 聆风clark
# @Time: 2021/11/8 5:53 下午
# @File: pbkdf2_demo.py
# @project demand:
import binascii
from Cryptodome.Hash import SHA1
from Cryptodome.Protocol.KDF import PBKDF2


text = 'I love Python!'
salt = b'43215678'
result = PBKDF2(text,  salt, count=10, hmac_hash_module=SHA1)
result = binascii.hexlify(result)
print(result)
# b'7fee6e8350cfe96314c76aaa6e853a50'

File: JavaScript-Reverse/Crypto/PBKDF2/pbkdf2_demo.js
// 引用 crypto-js 加密模块
var CryptoJS = require('crypto-js')

function pbkdf2Encrypt() {
    var text = "I love Python!"
    var salt = "43215678"
    // key 长度 128，10 次重复运算
    var encryptedData = CryptoJS.PBKDF2(text, salt, {keySize: 128 / 32, iterations: 10});
    return encryptedData.toString()
}

console.log(pbkdf2Encrypt())  // 7fee6e8350cfe96314c76aaa6e853a50
